src/algebra/lie/basic.lean

```diff
@@ -2013,6 +2091,10 @@ begin
   exact h₂ hI,
 end
 
+lemma semisimple_lie_abelian_trivial [is_semisimple R L] [h : is_lie_abelian L] :
+  (⊤ : lie_ideal R L) = ⊥ :=
```

Comment:
Is this the best way to express the fact that a Lie algebra is the zero Lie algebra? Should I have a separate predicate for this?

Reply:
I think we can use `subsingleton` for this. In principle the lemma could then become an `instance`, but we recently discovered that too many subsingleton instances slows down the library. So maybe it should stay a `lemma` for now.

Reply:
OK I'll leave it as it. I suppose the `subsingleton` approach would be to prove:
```lean
instance [is_semisimple R L] [h : is_lie_abelian L] : subsingleton (lie_algebra R L) :=
```
I think I can imagine ways in which this would make the typeclass inference engine's job a lot harder.

Reply:
I think @jcommelin is suggesting you use `subsingleton` without `instance`.

Reply:
Oh I see. I'll add that now.

Reply:
On reflection it is not clear to me I should create a subsingleton instance. Indeed IIUC, the statement:
```lean
lemma [is_semisimple R L] [is_lie_abelian L] : subsingleton (lie_algebra R L) :=
```
that I suggested above only demonstrates that two semisimple Abelian Lie algebras _with the same underlying Lie ring structures_ have the same module structure.

I guess I could also prove:
```lean
lemma [is_semisimple R L] [is_lie_abelian L] : subsingleton (lie_ring R L) :=
```
but I begin to wonder what all this does for me.

I'm happy to do it of course if you think I should.

Reply:
> Update: this is not quite totally trivial

can't you use `subsingleton_of_bot_eq_top` on your existing proof?

Reply:
The statement is
```lean
lemma semisimple_lie_abelian_trivial [is_semisimple R L] [h : is_lie_abelian L] : subsingleton (lie_ideal R L) :=
subsingleton_of_bot_eq_top (your proof)
```

Reply:
Oh I see; mathematically that's weaker than what I had in mind but will be far more useful. Thanks!